internal/operators/localmanager/localmanager.go

```diff
@@ -0,0 +1,356 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localmanager
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/operators"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+	runtimeclient "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/runtime-client"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	localgadgetmanager "github.com/inspektor-gadget/inspektor-gadget/pkg/local-gadget-manager"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	OperatorName  = "LocalManager"
+	Runtimes      = "runtimes"
+	ContainerName = "containername"
+)
+
+type MountNsMapSetter interface {
+	SetMountNsMap(*ebpf.Map)
+}
+
+type ContainersMapSetter interface {
+	SetContainersMap(*ebpf.Map)
+}
+
+type Attacher interface {
+	AttachGeneric(container *containercollection.Container, handler any) error
+	DetachGeneric(*containercollection.Container) error
+}
+
+type LocalManager struct {
+	localGadgetManager *localgadgetmanager.LocalGadgetManager
+	rc                 []*containerutils.RuntimeConfig
+}
+
+func (l *LocalManager) Name() string {
+	return OperatorName
+}
+
+func (l *LocalManager) Description() string {
+	return "Handles enrichment of container data and attaching/detaching to and from containers"
+}
+
+func (l *LocalManager) Dependencies() []string {
+	return nil
+}
+
+func (l *LocalManager) Params() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:          Runtimes,
+			Alias:        "r",
+			DefaultValue: strings.Join(containerutils.AvailableRuntimes, ","),
+			Description: fmt.Sprintf("Container runtimes to be used separated by comma. Supported values are: %s",
+				strings.Join(containerutils.AvailableRuntimes, ", ")),
+			IsMandatory: true,
+			// PossibleValues: containerutils.AvailableRuntimes, // TODO
+		},
+	}
+}
+
+func (l *LocalManager) PerGadgetParams() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:         ContainerName,
+			Alias:       "c",
+			Description: "Show only data from containers with that name",
+			IsMandatory: false,
+		},
+	}
+}
+
+func (l *LocalManager) CanOperateOn(gadget gadgets.Gadget) bool {
+	// We need to be able to get MountNSID and set ContainerInfo, so check for that first
+	_, canEnrichEvent := gadget.EventPrototype().(operators.KubernetesFromMountNSID)
+
+	// Secondly, we need to be able to inject the ebpf map onto the tracer
+	gi, ok := gadget.(gadgets.GadgetInstantiate)
+	if !ok {
+		log.Printf("cannot instantiate")
+		return false
+	}
+
+	instance, err := gi.NewInstance(nil)
```

Comment:
What do you think about having a `TracerPrototype` (or similar) method for this case? Then we could remove all checks for `runner == nil` in different implementations of `NewInstance` around.

Reply:
Hmm, the `runner == nil` check is something I really dislike as it's something you'd have to know about when writing a new gadget and isn't clear why it's needed. Another function like TracerPrototype seems redundant, but at least it's cleaner.

Two other options I could think of:
* handing over a prototype runner with gadget params set to their defaults (so they're at least present and won't lead to crashes) and a dummy/default logger
* adding another function like Prepare(runner)/PreRun(runner), which actually gets handed over the runner (that func could be part of the `gadgets.GadgetInstance` interface)

I'm not sure what the best way would be, but I'm leaning towards the Prepare()/PreRun(), if we're already adding a new function. It would have the added advantage of having at least one function in the `gadgets.GadgetInstance` interface, so it might be little bit clearer what to pass there.

Reply:
I don't follow your idea. What would be the task of this `Prepare(runner)/PreRun(runner)` function?

Reply:
I think it makes sense. I'm not sure I fully follow your example but removing the `runner == nil` check will help IMO.

Reply:
This has been solved by adding `Init(gadgetCtx)` to the gadget interface.